runtime/tests/interpreter/for_test.go

```diff
@@ -219,3 +219,34 @@ func TestInterpretForStatementEmpty(t *testing.T) {
 		value,
 	)
 }
+
+func TestInterpretForStatementCapturing(t *testing.T) {
+
+	t.Parallel()
+
+	inter := parseCheckAndInterpret(t, `
+       fun test(): Int {
+           let fs: [((): Int)] = []
+           for x in [1, 2, 3] {
+               fs.append(fun (): Int {
+                   return x
+               })
+           }
+           var sum = 0
+           for f in fs {
+              sum = sum + f()
```

Comment:
maybe push each function call's result to an output array, which ensures the correct ordering of captures too. this test will pass if 1 and 2 are discarded but the function returning 3 is run twice, for instance.

Reply:
Good idea! Improved the tests in [e4478d3](https://github.com/onflow/cadence/pull/2024/commits/e4478d3ac06134975b6f473edee095ae00718372)